### PR TITLE
Candidate for 4.0.35

### PR DESCRIPTION
**Pull request by kushti**
* #1765
* #1767 
* #1772 
* #1776

### CODE REVIEW

**Review by aslesarenko — CHANGES_REQUESTED** [review]

File: src/main/scala/scorex/core/network/peer/RestApiUrlPeerFeature.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+object RestApiUrlPeerFeatureSerializer extends ScorexSerializer[RestApiUrlPeerFeature] {
+
+  override def serialize(obj: RestApiUrlPeerFeature, w: Writer): Unit = {
+    val addr = obj.restApiUrl.toString.getBytes("UTF-8")
+    w.put(addr.size.toByteExact)
+    w.putBytes(addr)
+  }
+
+  override def parse(r: Reader): RestApiUrlPeerFeature = {
+    val fas = r.getUByte()
```
[the remark this answers is not in the document]
Comment (pragmaxim) [reply src/main/scala/scorex/core/network/peer/RestApiUrlPeerFeature.scala:34]: Addressed in https://github.com/ergoplatform/ergo/pull/1766/commits/5f6ea837c8e50ab0eb561219636720b920f94fd0 ... let's write it as unsinged byte and increase the max to 255 ... My apologies, this was a good catch, thank you

File: src/main/scala/org/ergoplatform/modifiers/BlockSection.scala [hunk 2]
```diff
@@ -1,30 +1,21 @@
...
-import scorex.util.{ModifierId, bytesToId, idToBytes}
+import io.circe.Encoder
+import org.ergoplatform.modifiers.history.extension.Extension
+import org.ergoplatform.modifiers.history.header.Header
+import org.ergoplatform.modifiers.history.{ADProofs, BlockTransactions}
+import scorex.core.PersistentNodeViewModifier
 
-/**
-  * An interface for Ergo block section which contains corresponding header id and a digest of its payload.
-  */
```
[the remark this answers is not in the document]
Comment (kushti) [reply src/main/scala/org/ergoplatform/modifiers/BlockSection.scala:10]: This is now a ScalaDoc for `NonHeaderBlockSection`, made a new ScalaDoc for BlockSection

File: src/main/scala/org/ergoplatform/settings/PeerFeatureIds.scala [hunk 3]
```diff
@@ -1,5 +1,26 @@
 package org.ergoplatform.settings
 
+/**
+  * Repository of existing peer feature identifiers
+  */
 object PeerFeatureIds {
+  /**
+    * See `LocalAddressPeerFeature`
```
[the remark this answers is not in the document]
Comment (kushti) [reply src/main/scala/org/ergoplatform/settings/PeerFeatureIds.scala:8]: It does

[the remark this answers is not in the document]
Comment (kushti) [reply src/main/scala/org/ergoplatform/settings/PeerFeatureIds.scala:8]: Fixed by @pragmaxim

File: src/main/scala/scorex/core/network/message/BasicMessagesRepo.scala [hunk 4]
```diff
@@ -249,12 +247,14 @@ object HandshakeSpec {
   * to the receiving node at the beginning of a connection. Until both peers
   * have exchanged `Handshake` messages, no other messages will be accepted.
   */
-class HandshakeSpec(featureSerializers: PeerFeature.Serializers, sizeLimit: Int) extends MessageSpecV1[Handshake] {
+class HandshakeSerializer(featureSerializers: PeerFeature.Serializers) extends MessageSpecV1[Handshake] {
+
+  private val maxHandshakeSize = 8096
```
[the remark this answers is not in the document]
Comment (kushti) [reply src/main/scala/scorex/core/network/message/BasicMessagesRepo.scala]: Not sure it is needed actually, no any uses around

**Review by pragmaxim — APPROVED** [review]

**Review by aslesarenko — APPROVED** [review]

**Pull request merged**